includes/sanitizers/class-amp-iframe-sanitizer.php

```diff
@@ -117,6 +117,19 @@ public function sanitize() {
 			$this->did_convert_elements = true;
 			if ( empty( $normalized_attributes['layout'] ) && ! empty( $normalized_attributes['width'] ) && ! empty( $normalized_attributes['height'] ) ) {
 				$normalized_attributes['layout'] = 'intrinsic';
+
+				// Set layout to responsive if the iframe is aligned to full width.
+				$figure_node = null;
+				if ( $node->parentNode instanceof DOMElement && 'figure' === $node->parentNode->tagName ) {
+					$figure_node = $node->parentNode;
+				}
+				if ( $node->parentNode->parentNode instanceof DOMElement && 'figure' === $node->parentNode->parentNode->tagName ) {
+					$figure_node = $node->parentNode->parentNode;
+				}
+				if ( $figure_node && $figure_node->hasAttribute( 'class' ) && in_array( 'alignfull', explode( ' ', $figure_node->getAttribute( 'class' ) ), true ) ) {
+					$normalized_attributes['layout'] = 'responsive';
+				}
+
```

Comment:
Would this change be good to cherry-pick into a separate PR with its own issue?

Reply:
=> https://github.com/ampproject/amp-wp/issues/3506